icu4c/source/common/uloc_tag.cpp

```diff
@@ -2246,6 +2246,7 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
                 var = (VariantListEntry*)uprv_malloc(sizeof(VariantListEntry));
                 if (var == NULL) {
                     *status = U_MEMORY_ALLOCATION_ERROR;
+                    uprv_free(pExtension);
```

Comment:
Since ExtensionListEntry is a struct and defined as
```
struct ExtensionListEntry : public icu::UMemory {
```
could we change this by changing 
```
    ExtensionListEntry *pExtension;
```
to
```
LocalMemory<ExtensionListEntry> pExtension;
```
see  third_party/icu/source/common/cmemory.h
so just let the LocalMemory to handle the free for us.

@markusicu  Is this correct?

Reply:
> @markusicu Is this correct?

Almost. Since it's a UMemory and thus uses new/delete, we should use LocalPointer (unicode/localpointer.h).
LocalMemory is for raw types where we use uprv_malloc/uprv_free.

Reply:
Something like @markusicu suggested seems to be already in current main.